### DISCUSSION TITLE
Lacking Multivariate Normal distribution

### DISCUSSION THREAD

**Discussion by don-reba** [mathnet/mathnet-numerics#784] [Ideas]
I was surprised to find Math.NET to lack such a common distribution. I understand that `MatrixNormal` can be adapted to generate it, with a bit of computational and syntactic overhead. However, this is only possible for positive definite covariance matrices, since `MatrixNormal` is implemented via a Cholesky decomposition. Positive semi-definite covariance matrices are common and useful. Could a Multivariate Normal decomposition be added? Or else, could `MatrixNormal` be extended to handle positive semi-definite covariances?

**Comment by cdrnet** [comment]
Yes, such a distribution could certainly be added. I'd accept such a contribution.

**Reply by don-reba** [reply]
Good to know. I'll look into contributing.

**Reply by don-reba** [reply]
Implemented it. Sent you an email about making a pull request.

**Reply by sdwfrost** [reply]
Hi, quick question on this - are there any updates on the PR?

**Reply by don-reba** [reply]
> Hi, quick question on this - are there any updates on the PR?

The fork is here: https://github.com/don-reba/mathnet-numerics. I used it for a personal project but have not received any feedback from Christoph regarding merging it to Mathnet Numerics.
